jetty-server/src/test/java/org/eclipse/jetty/server/BlockingContentProducerTest.java

```diff
@@ -116,6 +119,160 @@ public void testBlockingContentProducerNoInterceptorWithError()
         }
     }
 
+    @Test
+    public void testBlockingContentProducerInterceptorGeneratesError()
+    {
+        AtomicInteger contentSucceededCount = new AtomicInteger();
+        ContentProducer contentProducer = new BlockingContentProducer(new AsyncContentProducer(new StaticContentHttpChannel(new HttpInput.Content(ByteBuffer.allocate(1))
+        {
+            @Override
+            public void succeeded()
+            {
+                contentSucceededCount.incrementAndGet();
+            }
+        })));
+        try (AutoLock lock = contentProducer.lock())
+        {
+            contentProducer.setInterceptor(content -> new HttpInput.ErrorContent(new Throwable("testBlockingContentProducerInterceptorGeneratesError interceptor error")));
+
+            HttpInput.Content content1 = contentProducer.nextContent();
+            assertThat(content1.isSpecial(), Matchers.is(true));
```

Comment:
`import static org.hamcrest.Matchers.is;` like in `AsyncContentProducerTest` looks nicer

Reply:
Done, thanks.